api/server/structs/endpoints_validator.go

```diff
@@ -118,3 +118,27 @@ type GetValidatorPerformanceResponse struct {
 	MissingValidators             [][]byte `json:"missing_validators,omitempty"`
 	InactivityScores              []uint64 `json:"inactivity_scores,omitempty"`
 }
+
+type GetValidatorParticipationRequest struct {
+	Epoch   string `json:"epoch"`
+	StateID bool   `json:"state_id"`
```

Comment:
Looks like you do not ever read the `epoch` parameter in the handler and rely on `state_id` instead. This makes sense to me but:
- we have to remove the `epoch` param if we ignore it anyway; otherwise it will confuse callers
- let's handle `state_id` in the same way as in Beacon API handlers; this means using the `Stater` to fetch states, which allows us to pass the following into the param:
> "head" (canonical head in node's view), "genesis", "finalized", "justified", <slot>, <hex encoded stateRoot with 0x prefix>

Reply:
We do use the epoch, I set it to prioritize the `state_id` if it's set but it isn't required so in the default case where `state_id` doesn't match anything we set the epoch which is technically required in this case so that `else` I have doesn't actually do anything and would need to be removed here, or just remove the `required`:

```
	default:
		_, e, ok := shared.UintFromQuery(w, r, "epoch", true)
		if !ok {
			currentSlot := s.CoreService.GenesisTimeFetcher.CurrentSlot()
			currentEpoch := slots.ToEpoch(currentSlot)
			epoch = uint64(currentEpoch)
		} else {
			epoch = e
		}
	}
```

Reply:
Ah OK, I missed it. I don't have that strong of an opinion on this, but having just a `state_id` and using the `Stater` similarly to how we do it in Beacon API makes things simpler and more uniform.